### ISSUE TITLE
Screen readers does not announce certain states correctly for checkbox tree views

### ISSUE THREAD

**Issue by mellis481** [lissitz/react-accessible-treeview#32]
`aria-selected` and `aria-multiselectable` are used for all instances of the TreeView.  This results in a reasonable screen reader UX for TreeView instances which do not involve checkboxes when nodes can only be selected.  For TreeView instances which do include checkboxes, however, it creates a user experience which could be better.

Specifically, because of the `aria-selected` usage:
1. a checkbox node with indeterminate state (half selected) reads out "not selected".
2. the selected state of a selected node is not read out when you navigate to it (it only announces nodes that are not selected for some reason).

If `aria-checked` is used instead, the screen reader will read "checked" or "not checked" instead of "selected" or "not selected" which IMO more accurate describes the action executed.  More importantly, the use of `aria-checked` successfully:
1. handles the indeterminate state (using `aria-checked="mixed"`), reading "\<node name and level\> checkbox half checked".
3. announces the selected state of a selected node when you navigate to it. (eg. "avocados checked one of five level two").

The section of [this](https://www.w3.org/WAI/ARIA/apg/patterns/treeview#wai-aria-roles-states-and-properties-25) W3C article on making a treeview accessible which starts _"The selection state of each selectable node is indicated with either aria-selected or aria-checked"_ provides describes that `aria-selected` or `aria-checked` can be used in tree views.

A way to handle this which seems advisable to me is to provide a prop for the TreeView component to allow consumers to indicate if the TreeView will be using checkboxes or not (perhaps `nodeAction?: 'check' | 'select'`).  If a consumer indicates their TreeView instances uses checkboxes, the resulting HTML will use the `aria-checked` attributes; else, it will use the existing `aria-selected` and `aria-multiselectable` attributes. These changes have been submitted in PR https://github.com/lissitz/react-accessible-treeview/pull/33.
